lib/animina/accounts/resources/fast_user.ex

```diff
@@ -332,6 +374,46 @@ defmodule Animina.Accounts.FastUser do
       }
   end
 
+  defp build_query_for_credit_points(
```

Comment:
Why is this so complicated? Please add a comment to functions which are so complex why it has to be and what they are doing.